src/odemis/driver/orsay.py

```diff
@@ -1389,3 +1440,1097 @@ def terminate(self):
+                p = self._parameters[i]
+
+                lowerbound = None
+                if self._minpar is not None:  # in case a minimum parameter is supplied
+                    if self._minpar[i].Actual is not None:
+                        lowerbound = self._minpar[i].Actual
+                    else:
+                        lowerbound = self._minpar[i].Target
+                if lowerbound is None:
+                    lowerbound = p.Min
+                else:
+                    if p.Min is not None and not p.Min == lowerbound:
+                        raise AssertionError("%s.Min and %s contain different, non-None values."
+                                             "Contact Orsay Physics about this!" % (p.Name, self._minpar[i].Name))
+
+                if lowerbound is not None:  # if a lowerbound is defined in the server
+                    new_range[0][i] = self._parameter_to_VA_value(lowerbound)  # copy it to the va
+
+                upperbound = None
+                if self._maxpar is not None:  # in case a minimum parameter is supplied
+                    if self._maxpar[i].Actual is not None:
+                        upperbound = self._maxpar[i].Actual
+                    else:
+                        upperbound = self._maxpar[i].Target
+                if upperbound is None:
+                    upperbound = p.Max
+                else:
+                    if p.Max is not None and not p.Max == upperbound:
+                        raise AssertionError("%s.Max and %s contain different, non-None values."
+                                             "Contact Orsay Physics about this!" % (p.Name, self._maxpar[i].Name))
+                if upperbound is not None:  # if an upperbound is defined in the server
+                    new_range[1][i] = self._parameter_to_VA_value(upperbound)  # copy it to the va
+
+            if len(new_range[0]) == 1:
+                new_range = (new_range[0][0], new_range[1][0])
+            else:
+                new_range = (tuple(new_range[0]), tuple(new_range[1]))
+
+            self._va._value = new_range[0]
+            self._va.range = new_range
+            self._va.notify(new_range[0])
+
+        for p in self._parameters:
+            p.Subscribe(self.update_VA)
+
+        self.update_VA()
+
+    def disconnect(self):
+        """
+        Unsubscribes the VA from the parameter
+        """
+        if self._va is not None and self._parameters is not None:
+            for p in self._parameters:
+                p.Unsubscribe(self.update_VA)
+            self._parameters = None
+            self._attributeName = None
+            self._va._setter = WeakMethod(self._va._VigilantAttribute__default_setter)
+            self._va = None
+            self._va_type_name = None
+            self._conversion = None
+
+    def update_VA(self, parameter=None, attributeName=None):
+        """
+        parameter (Orsay Parameter): the parameter on the Orsay server that calls this callback
```

Comment:
Actually this is almost correct form of formatting. I you would like to explain the function you can add a line at the top so like this:
```
        Copies the value of the parameter to the VA
        parameter (Orsay Parameter): the parameter on the Orsay server that calls this callback
        attributeName (str): the name of the attribute of parameter which was changed
```